test/e2e/framework/network/utils.go

```diff
@@ -205,6 +210,8 @@ type NetworkingTestConfig struct {
 	// The kubernetes namespace within which all resources for this
 	// config are created
 	Namespace string
+	// Whether to prefer node External Addresses for the tests
+	PreferExternalAddresses bool
```

Comment:
I think @aojea's pattern suggestion is great.  However as written this still changes the default behavior for everything to prefer internal addresses by default.  

I would suggest we either not merge this until post-1.26, or you flip this around so it defaults to the existing behavior of preferring external addresses (and you can selectively change for certain tests until test freeze).

Reply:
I don't have any strong preferences to either but I've changed the code to just parameterize the option for now without change of behavior.